src/Application/Interfaces/IAppUserRepository.cs

```diff
@@ -1,14 +1,11 @@
-﻿using Domain.Entities;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿using Application.Pagination;
+using Domain.Entities;
 
 namespace Application.Interfaces
 {
     public interface IAppUserRepository : IGenericRepository<AppUser>
     {
         Task<AppUser> GetByTelegramId(long telegramId, bool tracking);
+        Task<PagedList<AppUser>> GetPagedAsync(PagedQueryBase query, bool tracking);
```

Comment:
I think you should create another generic interface with GetPagedAsync generic method

Reply:
like this?
![image](https://user-images.githubusercontent.com/67834974/157252904-1507718b-1b78-4967-a8ee-2f66b2dcbcd8.png)